hack/generator/pkg/astmodel/package_reference.go

```diff
@@ -70,3 +70,11 @@ func (pr *PackageReference) PackageName() string {
 func (pr *PackageReference) Equals(ref *PackageReference) bool {
 	return pr.packagePath == ref.packagePath
 }
+
+// String returns the string representation of the package reference
+func (pr *PackageReference) String() string {
+	return pr.packagePath
+}
+
+// Ensure we implement Stringer
+var _ fmt.Stringer = &PackageReference{}
```

Comment:
I've seen this more often as 
```suggestion
var _ fmt.Stringer = (*PackageReference)(nil)
```

... which avoids allocating. Obviously it isn't a huge deal since it's package level but can add up if you've got a big codebase!

Reply:
we had been doing that but someone pointed out that uneffectful discards like this will be removed by the compiler. will send you the link

Reply:
Yeah the thing I liked about this format was that it works the same if the receiver is not a ptr (just drop the `&`).